docs/mobile-apps/automated-testing/appium/test-app-upgrades.md

```diff
@@ -0,0 +1,210 @@
+```csharp
+// When using the file name
+driver.ExecuteScript("mobile: installApp", new Dictionary<string, string> { { "appPath", "storage:filename=<file-name>.apk|ipa" } });
+// When using the file id
+driver.ExecuteScript("mobile: installApp", new Dictionary<string, string> { { "appPath", "storage:<file-id>" } });
+```
+
+</TabItem>
+</Tabs>
+
+When installing apps mid-session, the app is not automatically launched. You will need to launch the app manually using the followings commands after installation.
+
+- **iOS:** `mobile: launchApp`
+- **Android:** `mobile: startActivity`
+
+For clarity we added two extra text lines in the Appium Commands list in the Sauce Labs UI to indicate that the app has been installed and needs to be launched manually.
+
+<img src={useBaseUrl('img/mobile-apps/appium-mid-session-logs.jpg')} alt="Mid session install logs" width="800" />
+
+The following commands will launch the newly installed app. Remember to replace the `bundleId` or `intent` with your specific application's identifier or main activity, respectively.
+
+<Tabs
+groupId="start-app"
+defaultValue="java"
+values={[
+{label: 'Java', value: 'java'},
+{label: 'Node.js', value: 'js'},
+{label: 'Python', value: 'python'},
+{label: 'Ruby', value: 'ruby'},
+{label: 'C#', value: 'csharp'},
+]}>
+
+<TabItem value="java">
+
+<!-- prettier-ignore -->
+```java
+// Android
+driver.executeScript("mobile: startActivity", ImmutableMap.of("intent", "com.saucelabs.mydemoapp.rn/.MainActivity"));
+// iOS
+driver.executeScript("mobile: launchApp", ImmutableMap.of("bundleId", "com.saucelabs.mydemoapp.rn"));
+```
+
+</TabItem>
+<TabItem value="js">
+
+<!-- prettier-ignore -->
+```js
+// Android
+driver.execute('mobile: startActivity', {intent: 'com.saucelabs.mydemoapp.rn/.MainActivity'});
+//  
+driver.execute('mobile: launchApp', { bundleId: 'com.saucelabs.mydemoapp.rn'});
+```
+
+</TabItem>
+<TabItem value="python">
+
+<!-- prettier-ignore -->
+```py
+# Android
+driver.execute_script("mobile: startActivity", {"intent": "com.saucelabs.mydemoapp.rn/.MainActivity"})
+# iOS
+driver.execute_script("mobile: launchApp", {"bundleId": "com.saucelabs.mydemoapp.rn"})
+```
+
+</TabItem>
+<TabItem value="ruby">
+
+<!-- prettier-ignore -->
+```ruby
+# Android
+@driver.execute_script("mobile: startActivity", {"intent" => "com.saucelabs.mydemoapp.rn/.MainActivity"})
+# iOS
+@driver.execute_script("mobile: launchApp", {"bundleId" => "com.saucelabs.mydemoapp.rn"})
+```
+
+</TabItem>
+<TabItem value="csharp">
+
+<!-- prettier-ignore -->
+```csharp
+// Android
+driver.ExecuteScript("mobile: startActivity", new Dictionary<string, string> { { "intent", "com.saucelabs.mydemoapp.rn/.MainActivity" } });
+// iOS
+driver.ExecuteScript("mobile: launchApp", new Dictionary<string, string> { { "bundleId", "com.saucelabs.mydemoapp.rn" } });
+```
+
+</TabItem>
+</Tabs>
+
+This is an example test execution that installs an app mid-session, launches it and:
+
+- validates that the version has been upgraded
```

Comment:
should we help them to validate if the right version is updated?
So if they call the upgrade what they think is version 1.5
Should we add a callback or can they validate what is the version of the app they run the test against? is it really 1.5?

Reply:
This will be released this week, thanks for the useful feedback